halibot/halauth.py

```diff
@@ -0,0 +1,77 @@
+import json
+import logging
+
+def hasPermission(perm, reply=False):
+	def real_dec(func):
+		def wrapper(self, msg, *args, **kwargs):
+			if self._hal.auth.hasPermission(msg.origin, msg.author, perm):
+				func(self, msg, *args, **kwargs)
+			elif reply:
+				self.reply(msg, body="Permission Denied")
+		return wrapper
+	return real_dec
+
+class HalAuth():
+
+	def __init__(self):
+		self.perms = []
+		self.enabled = False
+		self.log = logging.getLogger("Auth")
+
+	# Load permission file, and set to enabled
+	def load_perms(self, path):
+		self.path = path
+
+		try:
+			with open(self.path, "r") as f:
+				temp = json.loads(f.read())
+
+			# Roll back into triple, also technically validates format
+			self.perms = [(a,b,c) for a,b,c in temp]
+
+		except Exception as e:
+			self.log.error("Error loading permissions: {}".format(e))
+			# Return if can't find auth?
```

Comment:
I think we can safely assume that no permissions file should behave the same as an empty one.